cumulusci/utils.py

```diff
@@ -21,6 +22,23 @@
     os.path.join(os.path.dirname(os.path.realpath(__file__)), "..")
 )
 META_XML_CLEAN_DIRS = ("classes/", "triggers/", "pages/", "aura/", "components/")
+ZERO = timedelta(0)
+API_DATE_FORMAT = "%Y-%m-%dT%H:%M:%S.%f"
+DATETIME_LEN = (
+    4 + 1 + 2 + 1 + 2 + 1 + 2 + 1 + 2 + 1 + 2 + 1 + 3
+)  # the actual width of a datetime string formatted with above format
+
+
+def parse_api_datetime(value):
+    """ parse a datetime returned from the salesforce API.
+
+    in python 3 we should just use a strptime %z, but until then we're just going
+    to assert that its a fixed offset of +0000 since thats the observed behavior. getting
+    python 2 to support fixed offset parsing is too complicated for what we need imo."""
+    dt = datetime.strptime(value[0:DATETIME_LEN], API_DATE_FORMAT)
+    offset_str = value[DATETIME_LEN:]
+    assert offset_str == "+0000", "The Salesforce API returned a weird timezone."
```

Comment:
I've heard it recommended to not use `assert` in non-test code because the Python compiler removes assert statements when run with -O. Not that that's likely though...

Reply:
I guess that makes this just a silent failure if you're running with asserts off. Idk, I got the practice from DRF but I'm not tied to it.